Makefile

```diff
@@ -2,15 +2,56 @@ LIBRARY_NAME=libpubliccode-parser
 
 LIB_DIR=lib
 
-LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME).so
+# Detect current platform
+GOOS := $(shell go env GOOS)
+GOARCH := $(shell go env GOARCH)
 
-.PHONY: all build clean deps
+# Determine file extension based on OS
+ifeq ($(GOOS),darwin)
+	EXT=dylib
+else
+	EXT=so
+endif
+
+# Current platform library
+LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME)-$(GOOS)-$(GOARCH).$(EXT)
+
+# Legacy symlink for backward compatibility
+LEGACY_LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME).so
+
+.PHONY: all build build-all build-linux-amd64 build-linux-arm64 build-darwin-amd64 build-darwin-arm64 clean install
 
 all: build
 
+# Build for current platform
 build:
 	mkdir -p $(LIB_DIR)
 	cd go-src && CGO_ENABLED=1 go build -buildmode=c-shared -o ../$(LIBRARY) publiccode-parser-wrapper.go
+	@echo "Built $(LIBRARY)"
```

Comment:
We can remove the build of `publiccode-parser-wrapper.go`, because we never load it.